pandas/_libs/algos.pyx

```diff
@@ -96,22 +96,94 @@ class NegInfinity(object):
     __ge__ = lambda self, other: self is other
 
 
-cdef inline Py_ssize_t swap(numeric *a, numeric *b) nogil except -1:
-    cdef numeric t
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def is_lexsorted(list list_of_arrays):
+    cdef:
+        int i
+        Py_ssize_t n, nlevels
+        int64_t k, cur, pre
+        ndarray arr
+        bint result = 1
```

Comment:
We're not consistent on this, but I think using `True`/`False` on bint is clearer